### PR TITLE
Container manfiest tag edit form.

### PR DESCRIPTION
**Pull request by newswangerd**
## Changes
- Refactored execution environment detail pages so that the common logic between the pages is shared via the `withContainerRepo` higher order component.

## Demo
Screen Recording 2021-04-13 at 5 23 51 PM mov

### CODE REVIEW

**Comment by ZitaNemeckova** [comment]
Screenshot 2021-04-15 at 14 14 01
Clicking `Show Less` closes the modal.

**Comment by newswangerd** [comment]
>     * It's possible to remove `latest`. Is that ok?

Apparently it is. Red Hat has even removed the `latest` tag from a lot of the official Red Hat images: https://access.redhat.com/articles/4301321

File: src/utilities/parse-pulp-id.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const UUIDRegEx = /\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b/;
```
Comment (himdel) [inline src/utilities/parse-pulp-id.ts:1]: would `/\b[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\b/i` make sense? To catch UUID with capitals

File: src/containers/execution-environment-detail/tag-manifest-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+  };
+
+  private verifyTag(tag: string) {
+    if (!this.validateTagName(tag)) {
+      this.setState({
+        verifyingTag: false,
+        tagInFormError:
+          'A tag may contain lowercase and uppercase ASCII ' +
+          'alphabetic characters, digits, underscores, periods, and dashes. A tag must not ' +
+          'start with a period or a dash.',
```
Comment (himdel) [inline src/containers/execution-environment-detail/tag-manifest-modal.tsx:373]: Is this about VALID_TAG_REGEX? If so... the current regex doesn't allow leading underscores either.

Comment (newswangerd) [reply src/containers/execution-environment-detail/tag-manifest-modal.tsx:373]: Well, I copied the regex for this from pulp_container: https://github.com/pulp/pulp_container/blob/master/pulp_container/app/serializers.py#L31, so I'm not sure which one is right

File: src/containers/execution-environment-detail/execution_environment_detail_images.tsx [hunk 3]
```diff
@@ -79,48 +81,21 @@ class ExecutionEnvironmentDetailImages extends React.Component<
...
-        />
-        <Main>{this.renderImages()}</Main>
-      </React.Fragment>
-    );
+    return <Main>{this.renderImages()}</Main>;
   }
 
   renderImages() {
-    const { params, images } = this.state;
+    const { params, images, manageTagsMannifestDigest } = this.state;
```
Comment (himdel) [inline src/containers/execution-environment-detail/execution_environment_detail_images.tsx:98]: typo ma*nn*ifest

**Review by himdel — APPROVED** [review]
Everything seems to work :+1:, LGTM :)

**Pull request merged**